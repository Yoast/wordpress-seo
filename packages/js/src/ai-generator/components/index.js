export { App } from "./app";
export { GoogleContent } from "./google-content";
export { GooglePreview } from "./google-preview";
export { GooglePreviewSkeleton } from "./google-preview-skeleton";
export { Introduction } from "./introduction";
export { LengthProgressBar } from "./length-progress-bar";
export { ModalContent } from "./modal-content";
export { PreviewModePicker } from "./preview-mode-picker";
export { SocialContent } from "./social-content";
export { SocialPreview } from "./social-preview";
export { SocialPreviewSkeleton } from "./social-preview-skeleton";
export { SuggestionError } from "./suggestion-error";
export { SuggestionsList } from "./suggestions-list";
export { SuggestionsListSkeleton } from "./suggestions-list-skeleton";
export { TipNotification } from "./tip-notification";
export { TwitterContent } from "./twitter-content";
export { TwitterPreview } from "./twitter-preview";
export { TwitterPreviewSkeleton } from "./twitter-preview-skeleton";
export { TypeProvider } from "./type-provider";
export { SparksLimitNotification } from "./sparks-limit-notification";
