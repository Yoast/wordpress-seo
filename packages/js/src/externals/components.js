import CollapsibleCornerstone from "../components/CollapsibleCornerstone";
import KeywordInput from "../components/contentAnalysis/KeywordInput";
import ReadabilityAnalysis from "../components/contentAnalysis/ReadabilityAnalysis";
import SeoAnalysis from "../components/contentAnalysis/SeoAnalysis";

window.yoast = window.yoast || {};
window.yoast.externals = window.yoast.externals || {};
window.yoast.externals.components = {
	CollapsibleCornerstone,
	KeywordInput,
	ReadabilityAnalysis,
	SeoAnalysis,
};
