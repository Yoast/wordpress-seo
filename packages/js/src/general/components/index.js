export { AlertsList } from "./alerts-list";
export { AlertsTitle } from "./alerts-title";
export { Collapsible } from "./collapsible";
export { Notice } from "./notice";
export { Notifications } from "./notifications";
export { Problems } from "./problems";
export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
