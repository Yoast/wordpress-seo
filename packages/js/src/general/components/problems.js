import { __, _n } from "@wordpress/i18n";
import { useSelect } from "@wordpress/data";
import { ExclamationCircleIcon } from "@heroicons/react/outline";
import { Paper } from "@yoast/ui-library";
import { AlertsList } from "./alerts-list";
import { AlertsTitle } from "./alerts-title";
import { Collapsible } from "./collapsible";

import { AlertsContext } from "../contexts/alerts-context";
import { STORE_NAME } from "../constants/index";

/**
 * @returns {JSX.Element} The problems component.
 */
export const Problems = () => {
	const problemsList = useSelect( ( select ) => select( STORE_NAME ).selectActiveProblems(), [] );
	const dismissedProblemsList = useSelect( ( select ) => select( STORE_NAME ).selectDismissedProblems(), [] );

	const dismissedProblems = dismissedProblemsList.length;

	const dismissedProblemsLabel = _n(
		"hidden problem",
		"hidden problems",
		dismissedProblems,
		"wordpress-seo"
	);

	const problemsTheme = {
		Icon: ExclamationCircleIcon,
		bulletClass: "yst-fill-red-500",
		iconClass: "yst-text-red-500",
	};

	return (
		<Paper>
			<Paper.Content className="yst-max-w-[600px] yst-flex yst-flex-col yst-gap-y-6">
				<AlertsContext.Provider value={ { ...problemsTheme } }>
					<AlertsTitle title={ __( "Problems", "wordpress-seo" ) } counts={ problemsList.length }>
						<p className="yst-mt-2 yst-text-sm">
							{ problemsList.length > 0
								? __( "We have detected the following issues that affect the SEO of your site.", "wordpress-seo" )
								: __( "Good job! We could detect no serious SEO problems.", "wordpress-seo" )
							}
						</p>
					</AlertsTitle>
					<AlertsList items={ problemsList } />

					{ dismissedProblems > 0 && (
						<Collapsible label={ `${ dismissedProblems } ${ dismissedProblemsLabel }` }>
							<AlertsList className="yst-pb-6" items={ dismissedProblemsList } />
						</Collapsible>
					) }
				</AlertsContext.Provider>
			</Paper.Content>
		</Paper>
	);
};
