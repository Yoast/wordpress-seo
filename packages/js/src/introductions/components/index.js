export { Content } from "./content";
export { Introduction } from "./introduction";
export { Modal } from "./modal";
export { IntroductionProvider, useIntroductionsContext } from "./provider";
