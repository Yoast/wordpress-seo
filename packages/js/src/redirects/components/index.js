export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
export { Menu } from "./menu";
export { SidebarLayout } from "./sidebar-layout";
export { Notifications } from "./notifications";
