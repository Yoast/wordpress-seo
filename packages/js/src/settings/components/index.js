export { default as FieldsetLayout } from "./fieldset-layout";
export { default as FormLayout } from "./form-layout";
export { default as FormikFlippedToggleField } from "./formik-flipped-toggle-field";
export { default as FormikMediaSelectField } from "./formik-media-select-field";
export { default as FormikReplacementVariableEditorField } from "./formik-replacement-variable-editor-field";
export { default as FormikValueChangeField } from "./formik-value-change-field";
export { default as FormikWithErrorField } from "./formik-with-error-field";
export { default as Notifications } from "./notifications";
export { default as OpenGraphDisabledAlert } from "./open-graph-disabled-alert";
export { default as SidebarNavigation } from "./sidebar-navigation";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
