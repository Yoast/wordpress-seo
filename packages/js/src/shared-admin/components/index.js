export { AcademyUpsellCard } from "./academy-upsell-card";
export { AiGenerateTitlesAndDescriptionsUpsell } from "./ai-generate-titles-and-descriptions-upsell";
export { AiFixAssessmentsUpsell } from "./ai-fix-assessments-upsell";
export { ErrorFallback } from "./error-fallback";
export { FieldsetLayout } from "./fieldset-layout";
export { ReactComponent as G2LogoWhite } from "./g2-logo-white.svg";
export { MenuItemLink } from "./menu-item-link";
export { OutboundLink } from "./outbound-link";
export { PremiumUpsellCard } from "./premium-upsell-card";
export { PremiumUpsellList } from "./premium-upsell-list";
export { SidebarRecommendations } from "./sidebar-recommendations";
export { UnsavedChangesModal } from "./unsaved-changes-modal";
export { VideoFlow } from "./video-flow";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
export { SiteKitConsentModal } from "./site-kit-consent-modal";
