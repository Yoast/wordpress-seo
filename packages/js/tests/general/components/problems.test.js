import { render } from "../../test-utils";
import { Problems } from "../../../src/general/components/problems";

const items = [
	{
		id: "test-id-1",
		message: "You've added a new type of content. We recommend that you review the corresponding Search appearance settings.",
		type: "error",
		dismissed: false,
	},
	{
		id: "test-id-2",
		message: "We notice that you have installed WPML. To make sure your canonical URLs are set correctly, install and activate the WPML SEO add-on as well!",
		type: "error",
		dismissed: false,
	},
	{
		id: "test-id-3",
		message: "<b>Huge SEO Issue:</b> You're blocking access to robots. If you want search engines to show this site in their results, you must go to your Reading Settings and uncheck the box for Search Engine Visibility. I don't want this site to show in the search results.",
		type: "error",
		dismissed: false,
	},
	{
		id: "test-id-4",
		message: "We need to re-analyze some of your SEO data because of a change in the visibility of your taxonomies. Please help us do that by running the SEO data optimization. We estimate this will take less than a minute.",
		type: "error",
		dismissed: false,
	},
];

jest.mock( "@wordpress/data", () => ( {
	useSelect: jest.fn( () =>  items ),
	useDispatch: jest.fn( () => ( { toggleAlertStatus: jest.fn() } ) ),
} ) );

describe( "AlertsList", () => {
	it( "should match snapshot", () => {
		const { container } = render( <Problems /> );
		expect( container ).toMatchSnapshot();
	} );
} );
