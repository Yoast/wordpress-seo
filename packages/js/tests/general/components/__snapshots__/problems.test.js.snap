// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertsList should match snapshot 1`] = `
<div>
  <div
    class="yst-paper"
  >
    <div
      class="yst-paper__content yst-max-w-[600px] yst-flex yst-flex-col yst-gap-y-6"
    >
      <div>
        <div
          class="yst-flex yst-justify-between yst-gap-2 yst-items-center"
        >
          <svg
            aria-hidden="true"
            class="yst-w-6 yst-h-6 yst-text-red-500"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <h2
            class="yst-title yst-title--2 yst-grow"
          >
            Problems
             
            (4)
          </h2>
        </div>
        <p
          class="yst-mt-2 yst-text-sm"
        >
          We have detected the following issues that affect the SEO of your site.
        </p>
      </div>
      <ul
        class=""
      >
        <li
          class="yst-flex yst-justify-between yst-gap-x-5 yst-border-b yst-border-slate-200 last:yst-border-b-0 yst-py-6 first:yst-pt-0 last:yst-pb-0"
        >
          <div
            class="yst-mt-1"
          >
            <svg
              class="yst-fill-red-500"
              height="11"
              width="11"
            >
              <circle
                cx="5.5"
                cy="5.5"
                r="5.5"
              />
            </svg>
          </div>
          <div
            class="yst-text-sm yst-text-slate-600 yst-grow"
          >
            You've added a new type of content. We recommend that you review the corresponding Search appearance settings.
          </div>
          <button
            class="yst-button yst-button--secondary yst-button--small yst-self-center yst-h-8"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="yst-w-4 yst-h-4 yst-text-neutral-700"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </li>
        <li
          class="yst-flex yst-justify-between yst-gap-x-5 yst-border-b yst-border-slate-200 last:yst-border-b-0 yst-py-6 first:yst-pt-0 last:yst-pb-0"
        >
          <div
            class="yst-mt-1"
          >
            <svg
              class="yst-fill-red-500"
              height="11"
              width="11"
            >
              <circle
                cx="5.5"
                cy="5.5"
                r="5.5"
              />
            </svg>
          </div>
          <div
            class="yst-text-sm yst-text-slate-600 yst-grow"
          >
            We notice that you have installed WPML. To make sure your canonical URLs are set correctly, install and activate the WPML SEO add-on as well!
          </div>
          <button
            class="yst-button yst-button--secondary yst-button--small yst-self-center yst-h-8"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="yst-w-4 yst-h-4 yst-text-neutral-700"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </li>
        <li
          class="yst-flex yst-justify-between yst-gap-x-5 yst-border-b yst-border-slate-200 last:yst-border-b-0 yst-py-6 first:yst-pt-0 last:yst-pb-0"
        >
          <div
            class="yst-mt-1"
          >
            <svg
              class="yst-fill-red-500"
              height="11"
              width="11"
            >
              <circle
                cx="5.5"
                cy="5.5"
                r="5.5"
              />
            </svg>
          </div>
          <div
            class="yst-text-sm yst-text-slate-600 yst-grow"
          >
            <b>
              Huge SEO Issue:
            </b>
             You're blocking access to robots. If you want search engines to show this site in their results, you must go to your Reading Settings and uncheck the box for Search Engine Visibility. I don't want this site to show in the search results.
          </div>
          <button
            class="yst-button yst-button--secondary yst-button--small yst-self-center yst-h-8"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="yst-w-4 yst-h-4 yst-text-neutral-700"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </li>
        <li
          class="yst-flex yst-justify-between yst-gap-x-5 yst-border-b yst-border-slate-200 last:yst-border-b-0 yst-py-6 first:yst-pt-0 last:yst-pb-0"
        >
          <div
            class="yst-mt-1"
          >
            <svg
              class="yst-fill-red-500"
              height="11"
              width="11"
            >
              <circle
                cx="5.5"
                cy="5.5"
                r="5.5"
              />
            </svg>
          </div>
          <div
            class="yst-text-sm yst-text-slate-600 yst-grow"
          >
            We need to re-analyze some of your SEO data because of a change in the visibility of your taxonomies. Please help us do that by running the SEO data optimization. We estimate this will take less than a minute.
          </div>
          <button
            class="yst-button yst-button--secondary yst-button--small yst-self-center yst-h-8"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="yst-w-4 yst-h-4 yst-text-neutral-700"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="yst-shadow-sm yst-border-slate-300 yst-rounded-md yst-border"
      >
        <button
          aria-expanded="false"
          class="yst-w-full yst-flex yst-justify-between yst-py-4 yst-pe-4 yst-ps-6 yst-items-center"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r0:"
          type="button"
        >
          <div
            class="yst-font-medium"
          >
            4 hidden problems
          </div>
          <svg
            aria-hidden="true"
            class="yst-h-5 yst-w-5 flex-shrink-0 yst-text-slate-400"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 9l-7 7-7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
